scripts/common.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * common.php
+ *
+ * @created      11.10.2022
+ * @author       smiley <smiley@chillerlan.net>
+ * @copyright    2022 smiley
+ * @license      MIT
+ */
+
+if(PHP_MAJOR_VERSION < 7){
+	throw new RuntimeException('PHP 7+ required!');
+}
+
+if(!extension_loaded('curl')){
+	throw new RuntimeException('cURL extension not installed!');
+}
+
+if(!extension_loaded('openssl')){
+	throw new RuntimeException('OpenSSL extension not installed!');
+}
+
+
+define('ACTION_ROOT', $_SERVER['GITHUB_ACTION_PATH'] ?? realpath(__DIR__.'\\..'));
+define('WORKSPACE_ROOT', $_SERVER['GITHUB_WORKSPACE'] ?? ACTION_ROOT);
+define('ACTION_DOWNLOADS', WORKSPACE_ROOT.'\\downloads');
+define('SDK_BUILD_DEPS', realpath(WORKSPACE_ROOT.'\\..').'\\deps');
+
+if(!file_exists(ACTION_DOWNLOADS)){
+	mkdir(ACTION_DOWNLOADS);
```

Comment:
check if mkdir fails and exit with non-zero exit code or throw, maybe?

Reply:
The current directory choice is somewhat of a Notlösung as I'm using this to transfer information between the several sub-actions. Instead it should be in the action root of the actions toolkit i mentioned before, completely outside of any possible interferences. If it emits a warning there or errors out, this might be the least of your problems 🙃 . (I moved it into the workspace/.github dir for now https://github.com/cmb69/setup-php-sdk/pull/17/commits/ab9ac82fa06059cf1390eba82db224de90848fd5)